src/pudl/workspace/datastore.py

```diff
@@ -315,21 +352,30 @@ class Datastore:
 
     def __init__(
         self,
-        local_cache_path: Path | None = None,
-        cloud_cache_path: str | None = "s3://pudl.catalyst.coop/zenodo",
+        local_cache_path: str | Path | UPath | None = None,
+        cloud_cache_path: str | UPath | None = "s3://pudl.catalyst.coop/zenodo",
```

Comment:
Allows a variety of types to be used to initialize the Datastore. But they are all converted to `UPath` with an explicit protocol before being handed off to the `UPathCache` class.